packages/run-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -1,301 +1,328 @@
 // @ts-check
 
-import { E } from '@endo/eventual-send';
-import { Far } from '@endo/marshal';
-import { AssetKind, AmountMath, isNatValue } from '@agoric/ertp';
+import { AmountMath, isNatValue } from '@agoric/ertp';
 import { makeNotifierKit } from '@agoric/notifier';
 
 import {
   calcLiqValueToMint,
-  calcValueToRemove,
   calcSecondaryRequired,
+  calcValueToRemove,
 } from '@agoric/zoe/src/contractSupport/index.js';
 
 import '@agoric/zoe/exported.js';
+import { defineKindMulti } from '@agoric/vat-data';
+import { Far } from '@endo/marshal';
 import { makePriceAuthority } from './priceAuthority.js';
-import { makeSinglePool } from './singlePool.js';
-
-const { details: X } = assert;
+import { singlePool } from './singlePool.js';
 
 // Pools represent a single pool of liquidity. Price calculations and trading
 // happen in a wrapper class that knows whether the proposed trade involves a
 // single pool or multiple hops.
 
+export const publicPrices = prices => {
+  return { amountIn: prices.swapperGives, amountOut: prices.swapperGets };
+};
+
+/**
+ * @typedef {Readonly<{
+ * liquidityBrand: Brand<'nat'>,
+ * secondaryBrand: Brand<'nat'>,
+ * centralBrand: Brand<'nat'>,
+ * liquidityZcfMint: ZCFMint,
+ * liquidityIssuer: Issuer<any>,
+ * toCentralPriceAuthority: PriceAuthority,
+ * fromCentralPriceAuthority: PriceAuthority,
+ * quoteIssuerKit: IssuerKit,
+ * zcf: ZCF,
+ * timer: TimerService,
+ * paramAccessor,
+ * }>} ImmutableState
+ *
+ * @typedef {{
+ * updater: IterationObserver<any>,
+ * notifier: Notifier<any>,
+ * poolSeat: ZCFSeat,
+ * liqTokenSupply: bigint,
+ * }} MutableState
+ *
+ * @typedef {{
+ *   state: ImmutableState & MutableState,
+ *   facets: {
+ *     helper: import('@agoric/vat-data/src/types').KindFacet<typeof helperBehavior>,
+ *     pool: import('@agoric/vat-data/src/types').KindFacet<any>, // FIXME typeof poolBehavior
+ *     singlePool: import('@agoric/vat-data/src/types').KindFacet<VirtualPool>,
```

Comment:
`KindFacet` makes a facet from a behavior. `VirtualPool` is already a facet so,
```suggestion
 *     singlePool: VirtualPool,
```
or,
```
 *     singlePool: import('@agoric/vat-data/src/types').KindFacet<typeof singlePoolBehavior>,
```

Reply:
done